schematics/web/authentication/files/src/features/UserProvider/UserProvider.tsx.template

```diff
@@ -35,27 +35,32 @@ export const UserProvider: FC = ({ children }) => {
     } catch (e) {
       setUser(null);
     }
-  };
+  }, []);
+
+  const onHubCapsule: HubCallback = useCallback(
+    async ({ payload: { event } }) => {
+      switch (event as HubEvent) {
+        case HubEvent.SignIn:
+        case HubEvent.TokenRefresh:
+          await refreshUser();
+          break;
+        case HubEvent.SignOut:
+          setUser(null);
+          break;
+        default:
+          break;
+      }
+    },
+    []
+  );
```

Comment:
If you are not using that function anywhere else the you can contain it I useEffect and without using useCallback hook but that is not required so it's more your choice.